lib_static/redmine/i18n.rb

```diff
@@ -69,10 +69,27 @@ def localized_float(number, locale: ::I18n.locale, precision: 2)
       ("%.2f" % hours.to_f)
     end
 
-    def format_date(date)
-      return nil unless date
+    # Formats the given date or datetime as a date string according to the user's time zone
+    # and optional specified or system default format.
+    #
+    # @param date_or_time [Date|Time] The date or time object to format.
+    # @param time_zone [ActiveSupport::TimeZone] Use a different time zone than the current users's.
+    #   If provided, will output the time zone identifier
+    # @param format [String, nil] The strftime format to use for the date. If nil, the default
+    #   date format from `Setting.date_format` is used.
+    def format_date(date_or_time, time_zone: nil, format: Setting.date_format)
+      return nil unless date_or_time
+
+      local =
+        if time_zone
+          date_or_time.in_time_zone(time_zone).to_date
+        elsif date_or_time.is_a?(Date)
+          date_or_time
+        else
+          in_user_zone(time).to_date
+        end
 
-      Setting.date_format.blank? ? ::I18n.l(date.to_date) : date.strftime(Setting.date_format)
+      format.blank? ? ::I18n.l(local) : date.strftime(format)
```

Comment:
I recently found that when we use `strftime` with a format variable like `%B` (full month), `%b` (abbreviated month) or `%A` (weekday) those strings are always returned in English. We can also use `I18n.l` here as it supports the same format strings as `strftime` but does correctly translate those keys:

```suggestion
      format.blank? ? ::I18n.l(local) : ::I18n.l(local, format: format)
```

Also, you used the `date` variable here, this does not exist in the method

Reply:
@klaustopher sorry this was dormant for a while, I addressed your remarks 👍